odilo.py

```diff
@@ -0,0 +1,581 @@
+        url = self.RECORD_AVAILABILITY_ENDPOINT.replace('{recordId}', record_id)
+        status_code, headers, content = self.get(url)
+        content = json.loads(content)
+
+        if status_code == 200 and len(content) > 0:
+            return content
+        else:
+            self.log.warn('Cannot retrieve availability for record: ' + record_id + ' response http ' + status_code)
+            if content:
+                self.log.warn(content)
+            return None
+
+    @staticmethod
+    def _do_get(url, headers, **kwargs):
+        # More time please
+        if not kwargs:
+            kwargs = {"timeout": 60}
+        else:
+            kwargs['timeout'] = 60
+
+        if 'allow_redirects' not in kwargs:
+            kwargs['allow_redirects'] = True
+
+        response = HTTP.get_with_timeout(url, headers=headers, **kwargs)
+        return response.status_code, response.headers, response.content
+
+    @staticmethod
+    def _do_post(url, payload, headers, **kwargs):
+        # More time please
+        if not kwargs:
+            kwargs = {"timeout": 60}
+        else:
+            kwargs['timeout'] = 60
+
+        return HTTP.post_with_timeout(url, payload, headers=headers, **kwargs)
+
+
+class MockOdiloAPI(OdiloAPI):
+    @classmethod
+    def mock_collection(cls, _db):
+        library = DatabaseTest.make_default_library(_db)
+        collection, ignore = get_one_or_create(
+            _db, Collection,
+            name="Test Odilo Collection",
+            create_method_kwargs=dict(
+                external_account_id=u'library_id_123',
+            )
+        )
+        integration = collection.create_external_integration(
+            protocol=ExternalIntegration.ODILO
+        )
+        integration.password = u'abcdef123hijklm'
+        library.collections.append(collection)
+        return collection
+
+
+class OdiloRepresentationExtractor(object):
+    """Extract useful information from Odilo's JSON representations."""
+
+    log = logging.getLogger("OdiloRepresentationExtractor")
+
+    format_data_for_odilo_format = {
+        "ACSM": (
+            Representation.PDF_MEDIA_TYPE, DeliveryMechanism.ADOBE_DRM
+        ),
+        "PDF": (
+            Representation.PDF_MEDIA_TYPE, DeliveryMechanism.STREAMING_TEXT_CONTENT_TYPE
+        ),
+        "EPUB": (
+            Representation.EPUB_MEDIA_TYPE, DeliveryMechanism.STREAMING_TEXT_CONTENT_TYPE
+        ),
+        "MP3": (
+            Representation.MP3_MEDIA_TYPE, DeliveryMechanism.STREAMING_AUDIO_CONTENT_TYPE
+        ),
+        "MP4": (
+            Representation.MP4_MEDIA_TYPE, DeliveryMechanism.STREAMING_VIDEO_CONTENT_TYPE
+        ),
+        "WMV": (
+            Representation.WMV_MEDIA_TYPE, DeliveryMechanism.STREAMING_VIDEO_CONTENT_TYPE
+        ),
+        "JPG": (
+            Representation.JPEG_MEDIA_TYPE, DeliveryMechanism.NO_DRM
+        ),
+        "SCORM": (
+            Representation.ZIP_MEDIA_TYPE, DeliveryMechanism.NO_DRM
+        )
```

Comment:
This needs a more specific media type. An EPUB is a ZIP file but because it's served as `application/epub+zip` the client knows to render it as an ebook and not to open it in an archive manager.

I looked at the SCORM spec and it does not seem to define a media type, so I recommend we come up with one. How about `application/vnd.librarysimplified.scorm+zip`?

It may seem nitpicky to bring this up now, but if this isn't in place from the beginning you'll need to do an annoying database migration later, and if we never deal with it, a client will have no idea whether or not it can handle a given "ZIP" file, because so many things are packaged via ZIP.

Reply:
Ok, you are right. It sounds good, we could use the media type you suggest for the SCORM files.